src/transformStateWithClones.js

```diff
@@ -7,7 +7,40 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newState = { ...state };
+  const result = [];
+
+  for (const action of actions) {
+    const cloneState = {};
+
+    if (action.type === 'addProperties') {
+      addProperties(newState, action.extraData);
+    } else if (action.type === 'removeProperties') {
+      removeProperties(newState, action.keysToRemove);
+    } else if (action.type === 'clear') {
+      for (const prop in newState) {
+        delete newState[prop];
+      }
+    }
+
+    result.push(Object.assign(cloneState, newState));
```

Comment:
i think, you can write `result.push({ ...newState });` and don't cteate an extra object cloneState